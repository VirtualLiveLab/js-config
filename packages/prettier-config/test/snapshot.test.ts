import * as prettier from "prettier";
import { describe, expect, it } from "vitest";

describe("Test Prettier config with snapshot", () => {
  describe("normal preset", () => {
    it("matches snapshot", async () => {
      const options = await prettier.resolveConfig("index.ts", {
        config: `${import.meta.dirname}/../dist/index.js`,
      });

      expect(options).toMatchSnapshot();
    });
  });

  describe("astro preset", () => {
    it("matches snapshot with normal file", async () => {
      const options = await prettier.resolveConfig("index.ts", {
        config: `${import.meta.dirname}/../dist/astro.js`,
      });

      expect(options).toMatchSnapshot();
    });

    it("matches snapshot with astro file", async () => {
      const options = await prettier.resolveConfig("index.astro", {
        config: `${import.meta.dirname}/../dist/astro.js`,
      });

      expect(options).toMatchSnapshot();
    });
  });
});
