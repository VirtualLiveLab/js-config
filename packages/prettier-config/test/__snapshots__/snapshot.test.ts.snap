// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test Prettier config with snapshot > astro preset > matches snapshot with astro file 1`] = `
{
  "arrowParens": "always",
  "endOfLine": "lf",
  "jsxSingleQuote": false,
  "parser": "astro",
  "plugins": [
    "prettier-plugin-astro",
  ],
  "quoteProps": "consistent",
  "semi": true,
  "singleQuote": false,
  "tabWidth": 2,
  "trailingComma": "all",
}
`;

exports[`Test Prettier config with snapshot > astro preset > matches snapshot with normal file 1`] = `
{
  "arrowParens": "always",
  "endOfLine": "lf",
  "jsxSingleQuote": false,
  "plugins": [
    "prettier-plugin-astro",
  ],
  "quoteProps": "consistent",
  "semi": true,
  "singleQuote": false,
  "tabWidth": 2,
  "trailingComma": "all",
}
`;

exports[`Test Prettier config with snapshot > normal preset > matches snapshot 1`] = `
{
  "arrowParens": "always",
  "endOfLine": "lf",
  "jsxSingleQuote": false,
  "quoteProps": "consistent",
  "semi": true,
  "singleQuote": false,
  "tabWidth": 2,
  "trailingComma": "all",
}
`;
